packages/editor/src/components/media-placeholder/index.js

```diff
@@ -161,11 +164,51 @@ class MediaPlaceholder extends Component {
 					allowedTypes={ allowedTypes }
 					value={ value.id }
 					render={ ( { open } ) => (
-						<Button isLarge onClick={ open }>
+						<Button
+							isLarge
+							className="editor-media-placeholder__button"
+							onClick={ open }
+						>
 							{ __( 'Media Library' ) }
 						</Button>
 					) }
 				/>
+				{ onSelectURL && (
+					<div
+						className="editor-media-placeholder__url-input-container"
+					>
+						<Button
+							className="editor-media-placeholder__button"
+							onClick={ toggleURLInput }
+							isToggled={ isURLInputVisible }
+							isLarge>
+							{ __( 'Insert from URL' ) }
+						</Button>
+						{ isURLInputVisible && (
+							<URLPopover onClose={ this.closeURLInput }>
+								<form
+									className="editor-media-placeholder__url-input-form"
+									onSubmit={ this.onSubmitSrc }
+								>
+									<input
+										className="editor-media-placeholder__url-input-field"
+										type="url"
+										aria-label={ __( 'URL' ) }
+										placeholder={ __( 'Paste or type URL' ) }
+										onChange={ this.onChangeSrc }
+										value={ this.state.src }
+									/>
+									<IconButton
+										className="editor-media-placeholder__url-input-submit-button"
+										icon="editor-break"
+										label={ __( 'Apply' ) }
+										type="submit"
+									/>
+								</form>
+							</URLPopover>
+						) }
+					</div>
```

Comment:
This `render()` function is getting pretty hectic. Would it be difficult to pull e.g. `.editor-media-placeholder__url-input-container` out into its own component?

Reply:
I'm always happy seeing when someone wants to extract a new component 👍

Reply:
Moved it into a component in the same file for now - I imagine it'll change a bit when the rest of the url popover stuff is refactored.